File: app/Models/Customer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Customer extends Model
{
    use HasFactory;

    protected $fillable = ['name', 'email', 'phone', 'address', 'credit_limit', 'credit_balance', 'is_active', 'active', 'code'];

    protected $casts = [
        'credit_limit' => 'decimal:2',
        'credit_balance' => 'decimal:2',
        'is_active' => 'boolean',
    ];

    public function transactions(): HasMany
    {
        return $this->hasMany(Transaction::class);
    }
}

File: POSQUPRO-MOBILE/lib/models/customer.dart
class Customer {
  final int id;
  final String name;
  final String? email;
  final String? phone;
  final String? address;
  final double creditLimit;
  final bool isActive;

  Customer({
    required this.id,
    required this.name,
    this.email,
    this.phone,
    this.address,
    this.creditLimit = 0,
    this.isActive = true,
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      email: json['email'],
      phone: json['phone'],
      address: json['address'],
      creditLimit: double.tryParse(json['credit_limit']?.toString() ?? '0') ?? 0,
      isActive: json['is_active'] ?? true,
    );
  }
}
